iceoryx_hoofs/test/test.hpp

```diff
@@ -21,4 +21,9 @@
 #include <gtest/gtest.h>
 #include <memory>
 
+// defensive checks (IOX_ASSUME, IOX_PRECONDITION) are active during testing
+/// @todo this is preferably configured in cmake
+#define IOX_CHECK_PRECONDITIONS
+#define IOX_CHECK_ASSUMPTIONS
+
```

Comment:
Please remove this from here. It is easily forgotten when we add a cmake configuration. Just use what is defined in `hoofs/error_reporting.hpp` else we might run into trouble in the future.

I would also move away from the `define` and create a `constexpr bool IOX_CHECK_PRECONDITIONS {true};` then it can more easily be configured via `iceoryx_hoofs/cmake/iceoryx_hoofs_deployment.hpp.in`

Reply:
Configured in `configuration.hpp` now which contains a bunch (two ...) compile time constants and can be extended as needed.